### PR TITLE
chore: enhance unified and passthrough logs with additional context

### PR DESCRIPTION
**Pull request by paulkr**

### CODE REVIEW

**Review by sagoez — APPROVED** [review]

File: integrationos-unified/src/unified.rs [hunk 1]
```diff
@@ -366,7 +370,7 @@ impl UnifiedDestination {
                     "Mapping request body {}\nUsing js {js}",
                     serde_json::to_string_pretty(&body)
                         .map_err(|e| {
-                            error!("Could not convert body to pretty string {body:?}: {e}");
+                            error!("Failed to convert body to pretty string for connection model. ID: {}, Body: {}, Error: {}", config.id, body, e);
```
Comment (moekatib) [inline integrationos-unified/src/unified.rs:373]: Security concern: The error logging in here may expose sensitive user data by logging the full request body. We should modify this to log only non-sensitive information.

File: integrationos-unified/src/unified.rs [hunk 2]
```diff
@@ -400,7 +407,7 @@ impl UnifiedDestination {
                     "Mapped body to {}",
                     serde_json::to_string_pretty(&body)
                         .map_err(|e| {
-                            error!("Could not convert body to pretty string {body:?}: {e}");
+                            error!("Failed to convert mapped body to pretty string. ID: {}, Body: {}, Error: {}", config.id, body, e);
```
Comment (moekatib) [inline integrationos-unified/src/unified.rs:410]: Same as above.

File: integrationos-unified/src/unified.rs [hunk 3]
```diff
@@ -472,16 +482,16 @@ impl UnifiedDestination {
                     "Mapping request crud {}\nUsing js {js}",
                     serde_json::to_string_pretty(&request)
                         .map_err(|e| {
-                            error!(
-                                "Could not convert request crud to pretty string {request:?}: {e}"
-                            );
+                            error!("Failed to convert request crud to pretty string. ID: {}, Request: {:?}, Error: {}", config.id, request, e);
```
Comment (moekatib) [inline integrationos-unified/src/unified.rs:485]: Same here with the request.

File: integrationos-unified/src/unified.rs [hunk 4]
```diff
@@ -493,7 +503,7 @@ impl UnifiedDestination {
                     "Mapped request crud to {}",
                     serde_json::to_string_pretty(&res)
                         .map_err(|e| {
-                            error!("Could not convert crud to pretty string {res:?}: {e}");
+                            error!("Failed to convert crud to pretty string. ID: {}, Res: {:?}, Error: {}", config.id, res, e);
```
Comment (moekatib) [inline integrationos-unified/src/unified.rs:506]: ✚

File: integrationos-unified/src/unified.rs [hunk 5]
```diff
@@ -534,7 +544,7 @@ impl UnifiedDestination {
                     "Mapped request body to {path}: {}",
                     serde_json::to_string_pretty(&body)
                         .map_err(|e| {
-                            error!("Could not convert mapped body to pretty string {body:?}: {e}");
+                            error!("Failed to convert mapped body to pretty string. ID: {}, Body: {:?}, Error: {}", config.id, body, e);
```
Comment (moekatib) [inline integrationos-unified/src/unified.rs:547]: ✚

File: integrationos-unified/src/unified.rs [hunk 6]
```diff
@@ -607,7 +626,10 @@ impl UnifiedDestination {
             "Received response body: {}",
             serde_json::to_string_pretty(&body)
                 .map_err(|e| {
-                    error!("Could not convert mapped body to pretty string {body:?}: {e}");
+                    error!(
+                        "Failed to convert body to pretty string. ID: {}, Body: {:?}, Error: {} ",
+                        config.id, body, e
```
Comment (moekatib) [inline integrationos-unified/src/unified.rs:631]: ✚

File: integrationos-unified/src/unified.rs [hunk 7]
```diff
@@ -667,7 +691,7 @@ impl UnifiedDestination {
                     debug!(
                         "Mapping response crud {}\nUsing js {js}",
                         serde_json::to_string_pretty(&res_to_map).map_err(|e| {
-                            error!("Could not convert response crud to pretty string {res_to_map:?}: {e}");
+                            error!("Failed to convert response crud to pretty string. ID: {}, Response to Map: {:?}, Error: {}", config.id, res_to_map, e);
```
Comment (moekatib) [inline integrationos-unified/src/unified.rs:694]: ✚

File: integrationos-unified/src/unified.rs [hunk 8]
```diff
@@ -687,7 +711,8 @@ impl UnifiedDestination {
                     debug!(
                         "Mapped response crud to {}",
                         serde_json::to_string_pretty(&res).map_err(|e| {
-                            error!("Could not convert crud to pretty string {res:?}: {e}");
+                            error!("Failed to convert response crud to pretty string. ID: {}, Response: {:?}, Error: {}", config.id, res, e);
+
```
Comment (moekatib) [inline integrationos-unified/src/unified.rs:715]: ✚

File: integrationos-unified/src/unified.rs [hunk 9]
```diff
@@ -715,7 +740,11 @@ impl UnifiedDestination {
             body = if let Some(body) = body {
                 let wrapped_body = json!({"body":body});
                 let mut bodies = jsonpath_lib::select(&wrapped_body, path).map_err(|e| {
-                    error!("Could not select body at response path {path}: {e}");
+                    error!(
+                        "Failed to select body at response path. ID {}, Path {}, Error {}",
+                        config.id, path, e
```
Comment (moekatib) [inline integrationos-unified/src/unified.rs:745]: Same with the `e` it could have user data.

**Review by moekatib — CHANGES_REQUESTED** [review]

**Pull request merged**